### PR TITLE
[sbt 2] Restore server tests

### PR DESCRIPTION
**Pull request by adpi2**
- Port the `BuildServerEvalReporter` to Scala 3
- Use `scalatest` to run the server tests
- Run and fix all `BuildServerTests` (BSP)
- Fix `ServerCompletionTests`
- Reactivate the server tests on the CI

### CODE REVIEW

File: main/src/main/scala/sbt/Defaults.scala [hunk 1]
```diff
@@ -2711,6 +2699,18 @@ object Defaults extends BuildCommon {
     }
   )
 
+  private inline def analysisStore: AnalysisStore = {
```
Comment (eed3si9n) [inline main/src/main/scala/sbt/Defaults.scala:2704]: An inlined task?

Comment (adpi2) [reply main/src/main/scala/sbt/Defaults.scala:2704]: Yay that allow me to refactor code in tasks without creating intermediate task. I can use `.value` even if I'm not in a `Def.task`.

**Review by eed3si9n — DISMISSED** [review]
Thanks!

**Comment by adpi2** [comment]
@eed3si9n I need a new approval to merge.

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**